src/zope/traversing/tests/test_publicationtraverse.py

```diff
@@ -134,13 +127,78 @@ def testTraverseRelativeURL(self):
 
     def testMissingSkin(self):
         ob = Content()
-        from zope.traversing.publicationtraverse import PublicationTraverser
         t = PublicationTraverser()
         request = TestRequest()
         self.assertRaises(
             NotFound, t.traversePath, request, ob, '/++skin++missingskin')
 
 
+    def test_traversePath_trailing_slash(self):
+        class Traverser(PublicationTraverser):
+            def __init__(self):
+                self.names = []
+
+            def traverseName(self, request, ob, name):
+                self.names.append(name)
+
+
+        t = Traverser()
+        t.traversePath(None, None, 'abc/def/')
+        self.assertEqual(t.names, ['abc', 'def'])
+
+        t = Traverser()
+        t.traversePath(None, None, 'abc/def///')
+
+        # Note that only *one* trailing slash is removed
```

Comment:
This could be considered a bug.

Reply:
Maybe mention that in the unit test comment?  Otherwise somebody will assume it's intentional -- "look, they even have a unit test for this".

Reply:
Good point.